service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/azure/container/CloneControlledAzureStorageContainerResourceFlight.java

```diff
@@ -0,0 +1,125 @@
+import bio.terra.workspace.service.resource.controlled.flight.create.GetCloudContextStep;
+import bio.terra.workspace.service.resource.controlled.flight.update.RetrieveControlledResourceMetadataStep;
+import bio.terra.workspace.service.resource.controlled.model.ControlledResource;
+import bio.terra.workspace.service.resource.model.CloningInstructions;
+import bio.terra.workspace.service.resource.model.WsmResourceType;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys;
+import bio.terra.workspace.service.workspace.model.CloudPlatform;
+import java.util.Optional;
+import java.util.UUID;
+
+public class CloneControlledAzureStorageContainerResourceFlight extends Flight {
+
+  public CloneControlledAzureStorageContainerResourceFlight(
+      FlightMap inputParameters, Object applicationContext) {
+    super(inputParameters, applicationContext);
+
+    FlightUtils.validateRequiredEntries(
+        inputParameters,
+        WorkspaceFlightMapKeys.ResourceKeys.RESOURCE,
+        JobMapKeys.AUTH_USER_INFO.getKeyName(),
+        WorkspaceFlightMapKeys.ControlledResourceKeys.CLONING_INSTRUCTIONS,
+        WorkspaceFlightMapKeys.ControlledResourceKeys.DESTINATION_RESOURCE_ID,
+        WorkspaceFlightMapKeys.ControlledResourceKeys.DESTINATION_WORKSPACE_ID);
+
+    var flightBeanBag = FlightBeanBag.getFromObject(applicationContext);
+    var sourceResource =
+        inputParameters.get(WorkspaceFlightMapKeys.ResourceKeys.RESOURCE, ControlledResource.class);
+    var userRequest =
+        inputParameters.get(JobMapKeys.AUTH_USER_INFO.getKeyName(), AuthenticatedUserRequest.class);
+    var destinationWorkspaceId =
+        inputParameters.get(
+            WorkspaceFlightMapKeys.ControlledResourceKeys.DESTINATION_WORKSPACE_ID, UUID.class);
+
+    boolean mergePolicies =
+        Optional.ofNullable(
+                inputParameters.get(WorkspaceFlightMapKeys.MERGE_POLICIES, Boolean.class))
+            .orElse(false);
+    var cloningInstructions =
+        Optional.ofNullable(
+                inputParameters.get(
+                    WorkspaceFlightMapKeys.ControlledResourceKeys.CLONING_INSTRUCTIONS,
+                    CloningInstructions.class))
+            .orElse(sourceResource.getCloningInstructions());
+
+    final ControlledAzureStorageContainerResource sourceContainer =
+        sourceResource.castByEnum(WsmResourceType.CONTROLLED_AZURE_STORAGE_CONTAINER);
+    if (CloningInstructions.COPY_NOTHING == cloningInstructions) {
+      addStep(new SetNoOpContainerCloneResponseStep(sourceContainer));
+      return;
+    }
+
+    // Get the cloud context and store it in the working map
+    addStep(
+        new GetCloudContextStep(
+            destinationWorkspaceId,
+            CloudPlatform.AZURE,
+            flightBeanBag.getGcpCloudContextService(),
+            flightBeanBag.getAzureCloudContextService()),
+        RetryRules.shortDatabase());
```

Comment:
How does one decide which RetryRules to use?

Reply:
See the documentation in the code [here](https://github.com/DataBiosphere/terra-workspace-manager/blob/6c85db7b50ddf77de641710d6e8cda6437f6c99b/service/src/main/java/bio/terra/workspace/common/utils/RetryRules.java#L20)